pkgs/by-name/sd/sdformat/package.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cli11,
+  cmake,
+  pkg-config,
+  tinyxml-2,
+  urdfdom,
+  boost,
+  libuuid,
+  doxygen,
+  gz-cmake,
+  python3Full,
+  gz-utils,
+  gz-math,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sdformat";
+  version = "15.3.0";
+
+  src = fetchFromGitHub {
+    owner = "gazebosim";
+    repo = "sdformat";
+    tag = "sdformat${lib.versions.major finalAttrs.version}_${finalAttrs.version}";
+    hash = "sha256-5EGAypmWiUHvGpAXTIWJi8ChWkafK1li1C0/C1GIfkA=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    gz-cmake
+    python3Full
+    gz-utils
+    gz-math
+  ];
+
+  buildInputs = [
+    boost
+    cli11
+    doxygen
+    libuuid
+    tinyxml-2
+    urdfdom
+  ];
+
+  cmakeFlags = [
+    "-DUSE_INTERNAL_URDF=ON"
+    "-DBUILD_TESTING=ON"
```

```suggestion
    (lib.cmakeBool "USE_INTERNAL_URDF" true)
    (lib.cmakeBool "BUILD_TESTING" finalAttrs.doCheck)
```

Also, you're missing the test phase where the checks are actually run